src/terms/ewald.jl

```diff
@@ -82,56 +82,45 @@ function energy_ewald(lattice, recip_lattice, charges, positions; η=nothing, fo
     max_exp_arg = -log(eps(T)) + 5  # add some wiggle room
     max_erfc_arg = sqrt(max_exp_arg)  # erfc(x) ~= exp(-x^2)/(sqrt(π)x) for large x
 
+    # Precomputing summation bounds from cutoffs.
+    # Let A be the lattice matrix, then if ||Ax|| <= R, 
+    # then xi = <ei, A^-1 Ax> = <A^-T ei, Ax> <= ||A^-T ei|| R.
+    #
+    # Reciprocal space:  ||A^-1 G|| / 2η ≤ max_erfc_arg
+    # Real space: ||A(ti - tj - R)|| * η ≤ max_erfc_arg
+    Glims = ceil.(Int, [norm(inv(recip_lattice')[:, i]) * max_erfc_arg * 2η  for i in 1:3])
+    poslims = [maximum(rj[i] - rk[i] for rj in positions for rk in positions) for i in 1:3]
+    Rlims = ceil.(Int, [norm(inv(lattice')[:, i]) * max_erfc_arg / η + poslims[i] for i in 1:3])
```

Comment:
Ooh clever with the poslims, I would have just ignored it :-)

Reply:
It's a bit of a mouth full. Also the lines are too long. Maybe restructure a bit?

Reply:
The ceil(Int, x) is also making problems with IntervalArithmetic (failing the test), I think you had a workaround for a similar case before, should this be explicitly overloaded?